packages/odyssey-storybook/src/components/odyssey-labs/DataComponents/dataFunctions.ts

```diff
@@ -0,0 +1,123 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+// import {
+//   DataTableGetDataType,
+//   DataTableOnReorderRowsType,
+// } from "@okta/odyssey-react-mui";
+import {
+  DataGetDataType,
+  DataOnReorderRowsType,
+} from "@okta/odyssey-react-mui/labs";
+import { Person } from "./personData";
+
+export const filterData = ({
+  data,
+  ...args
+}: {
+  data: Person[];
+} & DataGetDataType) => {
+  let filteredData = data;
+  const { search, filters, sort, page = 1, resultsPerPage = 20 } = args;
```

Comment:
Fee-fi-fo-fum, I smell code mutations.

We should remove `let` and completely rewrite this to use implicit returns. I can work with you on this later. Too much to fix in a PR review.